src/ui/molecules/networkGraph/ecs/system/spawnGraphSystem.ts

```diff
@@ -0,0 +1,94 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+/* eslint-disable @typescript-eslint/prefer-readonly-parameter-types */
+import type { Mesh, Scene } from '@babylonjs/core'
+import { Color3, MeshBuilder, StandardMaterial, Vector3 } from '@babylonjs/core'
+import { component, toComponent, useInit } from '@javelin/ecs'
+import type { Link, Node } from 'ngraph.graph'
+import { LinkComponent, MeshComponent, NodeComponent, PositionComponent } from '../schema'
+import type { SystemFactory, World, WorldTickData } from '../type'
+
+type MeshFactory<T> = (o: T, scene: Scene) => Mesh
+export type NodeMeshFactory = MeshFactory<Node>
+export type LinkMeshFactory = MeshFactory<Link>
+
+export type SpawnGraphSystemOptions = {
+  nodeMeshFactory: NodeMeshFactory
+  linkMeshFactory: LinkMeshFactory
+}
+
+const defaultNodeMeshFactory: NodeMeshFactory = (node: Node, scene: Scene): Mesh => {
+  const material = new StandardMaterial('material#node', scene)
+  material.diffuseColor = new Color3(1, 1, 0.7)
+  material.alpha = 0.9
+
+  const mesh = MeshBuilder.CreateSphere(
+    `node:sphere#${node.id}`,
+    {
+      diameter: 5,
+      segments: 2
+    },
+    scene
+  )
+  mesh.metadata = {
+    '@tyle': 'node',
+    nodeId: node.id as number
+  }
+  mesh.material = material
+
+  return mesh
+}
+
+const defaultLinkMeshFactory: LinkMeshFactory = (link: Link, scene: Scene): Mesh => {
+  const mesh = MeshBuilder.CreateLines(
+    `link:line#${link.id}`,
+    {
+      points: [new Vector3(0, 0, 0), new Vector3(0, 0, 0)],
+      updatable: true
+    },
+    scene
+  )
+  mesh.color = new Color3(0.3, 0.3, 0.4)
+
+  return mesh
+}
+
+/**
+ * The SpawnGraphSystem system manages the relation between the graph domain and its representation in the scene, 
+ * by applying creational functions that turns graph nodes and links into meshes.
+ */
+export const SpawnGraphSystem: SystemFactory<SpawnGraphSystemOptions> =
+  (options?: SpawnGraphSystemOptions) => (world: World) => {
+    if (useInit()) {
+      const nodeMeshFactory = options?.nodeMeshFactory ?? defaultNodeMeshFactory
+      const linkMeshFactory = options?.linkMeshFactory ?? defaultLinkMeshFactory
+      const { graph, scene }: WorldTickData = world.latestTickData
+
+      const nodes: Map<number, number> = new Map()
+      graph.forEachNode((node: Node) => {
+        const id = node.id as number
```

Comment:
The node identifier is specified by a union type: `type NodeId = string | number`. By convention, we decided to use numbers as identifier. It's not great, and a more solid approach should be considered, but it would require some design first. If you agree, I'suggest to postpone to another PR.

Reply:
Yep, could be part of a next work 👌